src/client.rs

```diff
@@ -55,18 +59,26 @@ impl ClientPlugin {
         world.resource_scope(|world, mut client: Mut<RenetClient>| {
             world.resource_scope(|world, mut entity_map: Mut<ServerEntityMap>| {
                 world.resource_scope(|world, replication_rules: Mut<ReplicationRules>| {
+                    let mut stats = world.remove_resource::<ReplicationStats>();
                     while let Some(message) = client.receive_message(REPLICATION_CHANNEL_ID) {
                         let end_pos: u64 = message.len().try_into().unwrap();
                         let mut cursor = Cursor::new(message);
 
+                        if let Some(ref mut rs) = stats {
```

Comment:
Let's use `&mut` instead, it's just more common.
Also could you use the same name?
```suggestion
                        if let Some(stats) = &mut stats {
```